docs/getting-started/package-infrastructure.md

```diff
@@ -0,0 +1,878 @@
+```
+
+```console
+curl -OL https://raw.githubusercontent.com/crossplane/crossplane/master/docs/snippets/package/azure/composition.yaml
+```
+
+</div>
+<div class="tab-pane fade" id="alibaba-tab-1" markdown="1">
+
+```yaml
+apiVersion: apiextensions.crossplane.io/v1alpha1
+kind: Composition
+metadata:
+  name: compositepostgresqlinstances.alibaba.database.example.org
+  labels:
+    provider: alibaba
+    guide: quickstart
+spec:
+  writeConnectionSecretsToNamespace: crossplane-system
+  compositeTypeRef:
+    apiVersion: database.example.org/v1alpha1
+    kind: CompositePostgreSQLInstance
+  resources:
+    - base:
+        apiVersion: database.alibaba.crossplane.io/v1alpha1
+        kind: RDSInstance
+        spec:
+          forProvider:
+            engine: PostgreSQL
+            engineVersion: "9.4"
+            dbInstanceClass: rds.pg.s1.small
+            securityIPList: "0.0.0.0/0"
+            masterUsername: "myuser"
+          writeConnectionSecretToRef:
+            namespace: crossplane-system
+      patches:
+        - fromFieldPath: "metadata.uid"
+          toFieldPath: "spec.writeConnectionSecretToRef.name"
+          transforms:
+            - type: string
+              string:
+                fmt: "%s-postgresql"
+        - fromFieldPath: "spec.parameters.storageGB"
+          toFieldPath: "spec.forProvider.dbInstanceStorageInGB"
+      connectionDetails:
+        - fromConnectionSecretKey: username
+        - fromConnectionSecretKey: password
+        - fromConnectionSecretKey: endpoint
+        - fromConnectionSecretKey: port
+```
+
+```console
+curl -OL https://raw.githubusercontent.com/crossplane/crossplane/master/docs/snippets/package/alibaba/composition.yaml
+```
+
+</div>
+</div>
+
+## Build and Push The Configuration
+
+Finally, we'll author our metadata file then build and push our configuration
+so that Crossplane users may install it.
+
+> Note that Crossplane pushes packages to an OCI registry - currently [Docker
+> Hub] by default. You may need to run `docker login` before you are able to
+> push a package.
+
+<ul class="nav nav-tabs">
+<li class="active"><a href="#aws-tab-1" data-toggle="tab">AWS (Default VPC)</a></li>
+<li><a href="#aws-new-tab-1" data-toggle="tab">AWS (New VPC)</a></li>
+<li><a href="#gcp-tab-1" data-toggle="tab">GCP</a></li>
+<li><a href="#azure-tab-1" data-toggle="tab">Azure</a></li>
+<li><a href="#alibaba-tab-1" data-toggle="tab">Alibaba</a></li>
+</ul>
+<br>
+<div class="tab-content">
+<div class="tab-pane fade in active" id="aws-tab-1" markdown="1">
+
+```yaml
+apiVersion: meta.pkg.crossplane.io/v1alpha1
+kind: Configuration
+metadata:
+  name: getting-started-with-aws
+  annotations:
+    guide: quickstart
+    provider: aws
+    vpc: default
+```
+
+```console
+curl -OL https://raw.githubusercontent.com/crossplane/crossplane/master/docs/snippets/package/aws/crossplane.yaml
+
+# Set this to the OCI registry you wish to use.
+REPO=my-package-repo
```

Comment:
should we note this is optional?

Reply:
Let me try tweak the wording. I don't think it's actually optional here; you at least need to specify a username for Dockerup - e.g. `crossplane` or `negz`, but you could also use this to specify a different registry.